src/query/executor.rs

```diff
@@ -63,54 +71,104 @@ where
     }
 }
 
-#[allow(unused)]
-pub fn start_query(
+pub fn execute(
     config: QueryConfig,
     gateway: Gateway,
     input: ByteArrStream,
 ) -> JoinHandle<QueryResult> {
+    match (config.query_type, config.field_type) {
```

Comment:
This is a bit repetitious, but I think that this is a reasonable amount of repetition to manage.